lib/web/ca_export.go

```diff
@@ -0,0 +1,122 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package web
+
+import (
+	"archive/zip"
+	"bytes"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+
+	"github.com/gravitational/teleport/lib/client"
+)
+
+// authExportPublic returns the CA Certs that can be used to set up a chain of trust which includes the current Teleport Cluster
+//
+// GET /webapi/sites/:site/auth/export?type=<auth type>
+// GET /webapi/auth/export?type=<auth type>
+func (h *Handler) authExportPublic(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+	if err := h.authExportPublicError(w, r, p); err != nil {
+		http.Error(w, err.Error(), trace.ErrorToCode(err))
+		return
+	}
+
+	// Success output handled by authExportPublicInternal.
+}
+
+// authExportPublicError implements authExportPublic, except it returns an error
+// in case of failure. Output is only written on success.
+func (h *Handler) authExportPublicError(w http.ResponseWriter, r *http.Request, p httprouter.Params) error {
+	err := rateLimitRequest(r, h.limiter)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	query := r.URL.Query()
+	caType := query.Get("type") // validated by ExportAllAuthorities
+	format := query.Get("format")
+
+	const formatZip = "zip"
+	if format != "" && format != formatZip {
+		return trace.BadParameter("unsupported format %q", format)
+	}
+
+	ctx := r.Context()
+	authorities, err := client.ExportAllAuthorities(
+		ctx,
+		h.GetProxyClient(),
+		client.ExportAuthoritiesRequest{
+			AuthType: caType,
+		},
+	)
+	if err != nil {
+		h.logger.DebugContext(ctx, "Failed to generate CA Certs", "error", err)
+		return trace.Wrap(err)
+	}
+
+	if format == formatZip {
+		return h.authExportPublicZip(w, r, authorities)
+	}
+	if l := len(authorities); l > 1 {
+		return trace.BadParameter("found %d authorities to export, use format=%s to export all", l, formatZip)
+	}
+
+	// ServeContent sets the correct headers: Content-Type, Content-Length and Accept-Ranges.
+	// It also handles the Range negotiation
+	reader := strings.NewReader(string(authorities[0].Data))
```

Comment:
```suggestion
	reader := bytes.NewReader(authorities[0].Data)
```
No need to allocate a string here, right?